creator-node/src/serviceRegistry.js

```diff
@@ -44,64 +52,198 @@ class ServiceRegistry {
+
+      await utils.timeout(retryTimeoutMs, false)
+    }
+
+    this.logInfo('URSM Registration completed')
   }
 
-  const discoveryProviderWhitelist = config.get('discoveryProviderWhitelist')
-    ? new Set(config.get('discoveryProviderWhitelist').split(','))
-    : null
-  const identityService = config.get('identityService')
-
-  const audiusLibs = new AudiusLibs({
-    ethWeb3Config: AudiusLibs.configEthWeb3(
-      config.get('ethTokenAddress'),
-      config.get('ethRegistryAddress'),
-      ethWeb3,
-      config.get('ethOwnerWallet')
-    ),
-    web3Config: {
-      registryAddress: config.get('dataRegistryAddress'),
-      useExternalWeb3: true,
-      externalWeb3Config: {
-        web3: dataWeb3,
-        ownerWallet: config.get('delegateOwnerWallet')
+  /**
+   * Initialize SnapbackSM
+   * Requires L1 identity
+   */
+  async _initSnapbackSM () {
+    this.snapbackSM = new SnapbackSM(this.nodeConfig, this.libs)
+
+    const retryTimeoutMs = 10000 // ms
+    while (true) {
+      try {
+        this.logInfo(`Attempting to init SnapbackSM on ${retryTimeoutMs}ms interval...`)
+
+        await this.snapbackSM.init()
+        break
+
+        // Swallow all init errors
+      } catch (e) {
+        this.logError(`_initSnapbackSM Error`, e)
       }
-    },
-    discoveryProviderConfig: AudiusLibs.configDiscoveryProvider(discoveryProviderWhitelist),
-    // If an identity service config is present, set up libs with the connection, otherwise do nothing
-    identityServiceConfig: identityService ? AudiusLibs.configIdentityService(identityService) : undefined,
-    isDebug: config.get('creatorNodeIsDebug')
-  })
-
-  await audiusLibs.init()
-  return audiusLibs
+
+      await utils.timeout(retryTimeoutMs, false)
+    }
+
+    this.logInfo(`SnapbackSM Init completed`)
+  }
+
+  /**
+   * Creates, initializes, and returns an audiusLibs instance
+   *
+   * Configures dataWeb3 to be internal to libs, logged in with delegatePrivateKey in order to write chain TX
+   */
+  async _initAudiusLibs () {
+    const ethWeb3 = await AudiusLibs.Utils.configureWeb3(
+      config.get('ethProviderUrl'),
+      config.get('ethNetworkId'),
+      /* requiresAccount */ false
+    )
+    if (!ethWeb3) {
+      throw new Error('Failed to init audiusLibs due to ethWeb3 configuration error')
+    }
+
+    const discoveryProviderWhitelist = config.get('discoveryProviderWhitelist')
+      ? new Set(config.get('discoveryProviderWhitelist').split(','))
+      : null
+    const identityService = config.get('identityService')
+
+    const audiusLibs = new AudiusLibs({
+      ethWeb3Config: AudiusLibs.configEthWeb3(
+        config.get('ethTokenAddress'),
+        config.get('ethRegistryAddress'),
+        ethWeb3,
+        config.get('ethOwnerWallet')
+      ),
+      web3Config: AudiusLibs.configInternalWeb3(
+        config.get('dataRegistryAddress'),
+        [config.get('dataProviderUrl')],
+        // TODO - formatting this private key here is not ideal
```

Comment:
cc @hareeshnagaraj

Reply:
unf idt we have an option here :/

Reply:
eh, i think there's a cleaner way to do this but will leave for now